diff --git a/paths/paths.go b/paths/paths.go
index 6c9c3fa6c..0a059389f 100644
--- a/paths/paths.go
+++ b/paths/paths.go
@@ -31,6 +31,9 @@ func DefaultTailscaledSocket() string {
 	if runtime.GOOS == "plan9" {
 		return "/srv/tailscaled.sock"
 	}
+	if runtime.GOOS == "linux" {
+		return "/data/adb/tailscale/tailscaled.sock"
+	}
 	switch distro.Get() {
 	case distro.Synology:
 		if distro.DSMVersion() == 6 {
diff --git a/paths/paths_unix.go b/paths/paths_unix.go
index d317921d5..a5e203146 100644
--- a/paths/paths_unix.go
+++ b/paths/paths_unix.go
@@ -26,7 +26,7 @@ func statePath() string {
 	}
 	switch runtime.GOOS {
 	case "linux", "illumos", "solaris":
-		return "/var/lib/tailscale/tailscaled.state"
+		return "/data/adb/tailscale/tailscaled.state"
 	case "freebsd", "openbsd":
 		return "/var/db/tailscale/tailscaled.state"
 	case "darwin":
diff --git a/version/version_checkformat.go b/version/version_checkformat.go
index 05a97d191..8bc49888f 100644
--- a/version/version_checkformat.go
+++ b/version/version_checkformat.go
@@ -5,13 +5,14 @@
 
 package version
 
-import "fmt"
+// import "fmt"
 
 func init() {
 	// For official Android builds using the tailscale_go toolchain,
 	// panic if the builder is screwed up and we fail to stamp a valid
 	// version string.
 	if !isValidLongWithTwoRepos(Long()) {
-		panic(fmt.Sprintf("malformed version.Long value %q", Long()))
+		// panic(fmt.Sprintf("malformed version.Long value %q", Long()))
+		return
 	}
 }
